jcommune-view/jcommune-web-view/src/main/webapp/resources/javascript/app/postDraft.js

```diff
@@ -13,405 +13,255 @@
+                            }, 3000);
+                        } else {
+                            self._popup.show($labelConnectionLost);
+                        }
+                    } else if (xhr.status == 403) {
+                        self._popup.show($labelNotLoggedInError);
+                    }
+                });
+        } else {
+            return $.Deferred(function (deferred) {
+                deferred.fail();
+            });
         }
-      });
 
-      page.postTextArea.blur(function () {
-        if (!postPressed) {
-          saveEvent();
-        } else {
-          postPressed = false;
+        function enoughData(draft) {
+            return (draft['bodyText'] && draft['bodyText'].length > 0);
         }
-      });
-    }
+    };
 
-    function initGlobalListeners() {
-      $(window).on('unload', function () {
-        saveEvent();
-      });
-      $(".btn-toolbar").mouseup(function () {
-        draftState.wereNoChangesAfterLastSave = false;
-        startTimer();
-      });
-    }
-  }
-
-  var draftState = new DraftState();
-  var pageObject = new ViewModel(Page, draftState);
-
-  if (pageObject.isDraftApplicable()) {
-    var intervalId;
-    var lastSavingDate;
-    var dateUpdateCounter = 0;
-    var dateUpdateInterval;
-    var prevSavedMilis = 0;
-    var postPressed = false;
-
-    $("<span id='counter' class='keymaps-caption pull-right'></span>").insertAfter("#editorBbCodeDiv");
-    if (pageObject.isDraftExists() && pageObject.isDraftExistedOnServerBeforePageWasLoaded()) {
-      prevSavedMilis = parseInt(Page.nOfSavedMillisHiddenInput.val());
-      var differenceMillis = parseInt($("#differenceMillis").val());
-      var difSeconds = Math.floor(differenceMillis / 1000);
-      if (difSeconds <= 60) {
-        dateUpdateCounter = Math.floor(difSeconds / 5);
-        startFiveSecondsInterval();
-      } else if (difSeconds > 60 && difSeconds <= 3600) {
-        dateUpdateCounter = Math.floor(difSeconds / 60);
-        $("#counter").text(composeMinuteLabel(dateUpdateCounter));
-        dateUpdateInterval = setInterval(function () {
-          minuteIntervalHandler();
-        }, 60000);
-      } else if (difSeconds > 3600 && difSeconds <= 86400) {
-        dateUpdateCounter = Math.floor(difSeconds / 3600);
-        $("#counter").text(composeHourLabel(dateUpdateCounter));
-        dateUpdateInterval = setInterval(function () {
-          hourIntervalHandler();
-        }, 3600000);
-      } else {
-        printDate(new Date(prevSavedMilis));
-      }
-    }
+    /**
+     * Determines whether the post was changed from last saving.
+     *
+     * @returns {boolean} whether this draft was changed
+     */
+    PostDraft.prototype.wasChanged = function () {
+        var currentState = this._getDraftState(),
+            previousState = this._lastSavedDraftState;
+
+        return currentState['bodyText'] !== previousState['bodyText'];
```

Comment:
Because the body can be pretty large, can we first check the length and only if matches compare the contents? I assume that should be much faster.